qt-models/divecomputerextradatamodel.cpp

```diff
@@ -20,7 +20,7 @@ void ExtraDataModel::clear()
 
 QVariant ExtraDataModel::data(const QModelIndex &index, int role) const
 {
-	struct extra_data *ed = get_dive_dc(&displayed_dive, dc_number)->extra_data;
+	struct extra_data *ed = dc ? dc->extra_data : nullptr;
```

Comment:
what happens to us if the `dc` pointer becomes invalid? Do we know that any operation that could reallocate our dive table clears the model? Specifically, I don't see any code that resets the `dc` variable to nullptr.